### ISSUE TITLE
setting https_proxy causes singularity to not be able to pull image 

### ISSUE THREAD

**Issue by frankwillmore** [apptainer/singularity#3582]
## Version of Singularity:
3.2.0 

### Expected behavior

singularity pull shub://vsoch/hello-world will retrieve container

### Actual behavior
```
FATAL:   Failed to get manifest from Shub: No response received from singularity hub
```
### Steps to reproduce behavior

Set environment variable 'https_proxy' to any value will cause this behavior. We ordinarily set this to point to our proxy server, but setting any value at all will cause the failure.

**Comment by jscook2345** [comment]
@frankwillmore 

It looks like you have two things here. I'm not sure I follow how they are related, so perhaps you could open a separate issue for one of them and edit things up a bit?

For the proxy issue, if you set that incorrectly, I would not expect pulls from singularity hub to work. Are you setting the value correctly?

**Comment by frankwillmore** [comment]
I was trying to be as specific as possible in describing our build and it got muddied a bit. We've since been able to ascribe the mentioned linking issue to cray's static linking, so I edited the original issue to reflect that it's now just a v3.2.0 vanilla build that generates the FATAL error. 

We set https_proxy (and HTTPS_PROXY) to point to https://proxy:3128 which is our local setting. Our login hosts can still access the internet with or without this setting, but one would need the proxy to connect from our compute nodes. I've just verified that our proxy is working correctly, and that I cannot pull the image from login node when this is set, and I cannot pull an image at all from compute.
